src/main/scala/uk/gov/homeoffice/drt/analytics/prediction/dump/PaxPredictionDump.scala

```diff
@@ -0,0 +1,162 @@
+package uk.gov.homeoffice.drt.analytics.prediction.dump
+
+import akka.Done
+import akka.stream.Materializer
+import akka.stream.scaladsl.{Sink, Source}
+import org.apache.spark.ml.regression.LinearRegressionModel
+import org.apache.spark.sql.SparkSession
+import org.joda.time.DateTimeZone
+import org.slf4j.LoggerFactory
+import uk.gov.homeoffice.drt.analytics.prediction.DataSet
+import uk.gov.homeoffice.drt.arrivals.Arrival
+import uk.gov.homeoffice.drt.ports.Terminals.Terminal
+import uk.gov.homeoffice.drt.ports._
+import uk.gov.homeoffice.drt.time.{LocalDate, SDate}
+
+import java.io.{File, FileWriter}
+import scala.concurrent.{ExecutionContext, Future}
+
+case class PaxPredictionDump(arrivalsForDate: (Terminal, LocalDate) => Future[Seq[Arrival]],
+                             dumpStats: (String, String) => Future[Done])
+                            (implicit ec: ExecutionContext, mat: Materializer) extends ModelPredictionsDump {
+  private val log = LoggerFactory.getLogger(getClass)
+
+  override def dumpDailyStats(dataSet: DataSet,
+                              withIndex: Iterable[(Double, Seq[String], Seq[Double], String)],
+                              model: LinearRegressionModel,
+                              port: String,
+                              terminal: String,
+                             )
+                             (implicit sparkSession: SparkSession): Future[Done] = {
+    log.info(s"Dumping daily stats for $terminal")
+    val csvHeader = Seq(
+      "Date",
+      "Terminal",
+      "Actual pax",
+      "Pred pax",
+      "Flights",
+      "Actual per flight",
+      "Predicted per flight",
+      "Actual % cap",
+      "Pred % cap",
+      "Pred diff %",
+      "Fcst pax",
+      "Fcst % cap",
+      "Fcst diff %").mkString(",")
+
+    val predictions = predictionsWithLabels(dataSet, withIndex, model)
+
+    capacityStats(predictions, Terminal(terminal), arrivalsForDate)
+      .flatMap { stats =>
+        log.info(s"Got ${stats.size} days of stats for $terminal")
+        val fileWriter = new FileWriter(new File(s"/tmp/pax-forecast-$port-$terminal.csv"))
```

Comment:
We could have config for location of file creation even if it is temporary ? If it small size may be not be required but it creates big file we can mount volume in prod and pre prod env and use the mounted volume for location .

Reply:
Good point. It's mostly for use when running locally as the cronjob pods are deleted as soon as they've finished. But I can make it so that it only writes to a file if you give it a location :+1: